### ISSUE TITLE
blueprint "not contains" not working ?

### ISSUE THREAD

**Issue by bahout** [balderdashy/sails#2703]
Hi, 
"not contains" seems not working

I have tried :
```
http://localhost:1338/api/need?where={"product":{"not":[{"contains":"ordi"}]}}
```
I have not expected result

and
```
http://localhost:1338/api/need?where={"product":{"not":{"contains":"ordi"}}}
```
I have an error 500

Mongo is used as database

**Comment by Dallas62** [comment]
on MySQL, first request (with the array):

```
Details:  Error: ER_PARSE_ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '[object Object])   LIMIT 30 OFFSET 0' at line 1
```

And the second (without the array) ignore the `not`.

others tests with `like` is the same and 
`?where={"name":{"not": {"like": "t%"}}}` OR 
`?where={"name":{"not": {"like": "%t"}}}` OR 
`?where={"name":{"not": {"like": "%t%"}}}` OR 
`?where={"name":{"like": "t%"}}` OR 
`?where={"name":{"like": "%t"}}` OR 
`?where={"name":{"like": "%t%"}}`
return:
```
[
    {
        "name": "NotVisible",
        "id": 1,
        "createdAt": "2015-02-23T18:49:53.000Z",
        "updatedAt": "2015-02-23T18:49:53.000Z"
    },
    {
        "name": "Pictures",
        "id": 2,
        "createdAt": "2015-02-23T18:49:53.000Z",
        "updatedAt": "2015-02-23T18:49:53.000Z"
    },
    {
        "name": "Videos",
        "id": 3,
        "createdAt": "2015-02-23T18:49:53.000Z",
        "updatedAt": "2015-02-23T18:49:53.000Z"
    },
    {
        "name": "Reports",
        "id": 4,
        "createdAt": "2015-02-23T18:49:53.000Z",
        "updatedAt": "2015-02-23T18:49:53.000Z"
    }
]
```
AND
`?where={"name":{"like": "t"}}` OR
?where={"name":{"not":{"like": "t"}}}
return expected result:
```
[]
```

**Comment by tjwebb** [comment]
Yea, it looks like the sql building logic is not expecting an object there. Does this only happen with the mysql adapter?

**Comment by jetsonjohn** [comment]
Same issue for me also. I want to search for records not contains the value. I am using mongodb.

I have tried the following
````
where={"name":{"not":[{"contains":"foo"}]}}
where={"name":{"not":{"contains":"foo"}}}
where={"name":{"!":[{"contains":"foo"}]}}
where={"name":{"!":{"contains":"foo"}}}
where={"name":{"not": {"like": "foo%"}}}//Error - Details:  MongoError: Can't canonicalize query: BadValue $not cannot have a regex
where={"name":{"!": {"like": "foo%"}}}//Error - Details:  MongoError: Can't canonicalize query: BadValue $not cannot have a regex
````

Is there progress on this or is there any workaround for this issue

**Comment by sailsbot** [comment]
Thanks for posting, @bahout.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Dallas62** [comment]
Ignore is not a solution when we are waiting for some fixes / answers. @mikermcneil

**Comment by cyrilchapon** [comment]
This is still not working..
